libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tooling/includeKotlinToolingMetadataInApk.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.tooling
+
+import com.android.build.gradle.AppExtension
+import org.gradle.api.Project
+
+internal fun Project.includeKotlinToolingMetadataInApk() {
+    plugins.withId("com.android.application") {
+        val buildKotlinToolingMetadataTask = buildKotlinToolingMetadataTask ?: return@withId
+        val android = project.extensions.getByName("android") as AppExtension
+        android.applicationVariants.configureEach { variant ->
+            variant.registerPostJavacGeneratedBytecode(
```

Comment:
Can we do this only for the release (non-debuggable) variants and should there be an opt-out flag?

Reply:
Very good point!
Done: https://github.com/JetBrains/kotlin/pull/4530/commits/04ec716de5aac041a9399bd681fd5822df6b7617